src/python/pants/engine/internals/build_files.py

```diff
@@ -470,4 +485,7 @@ async def get_dependencies_rule_application(
 
 
 def rules():
-    return collect_rules()
+    return (
+        *collect_rules(),
+        QueryRule(BuildFileSymbolsInfo, ()),
```

Comment:
Nah, it's just normally deeply hidden in the weeds. The rule graph wouldn't work without them. They are the entry points into the graph. I need it here, as there is nothing else pulling this rule into the graph so I can use it, so without the explicit `QueryRule` this rule would be pruned out of the graph.

https://github.com/pantsbuild/pants/blob/03a1c48048cba39e650fdb7a23cba053e24e4595/src/python/pants/init/engine_initializer.py#L313-L330

Reply:
A comment here explaining the expected caller might be good. Or, better if possible would be defining the query closer to where it is used.

Reply:
> A comment here explaining the expected caller might be good. Or, better if possible would be defining the query closer to where it is used.

I'll add a comment. Prefer not to move it, as it's used in more than one place and would make it more difficult to reason about where it would be declared then..